src/utils.ts

```diff
@@ -40,10 +40,122 @@ export const encodeVector3D = function (value: Vector3D): pb.Vector3 {
 };
 
 /** Convert a 3D Vector Protobuf Datatype to a POJO */
-export const decodeVector3D = function (proto: pb.Vector3): Vector3D {
+export const decodeVector3D = function (proto: common.Vector3): Vector3D {
   return {
     x: proto.getX(),
     y: proto.getY(),
     z: proto.getZ(),
   };
 };
+
```

Comment:
Most of the code below will go away once we migrate to protobuf-es since that library generate protobuf structures that are essentially plain objects. for now we need to have custom functions to convert from objects -> protos.